File: src/org/webfabric/web/servlet/OriginalServletPath.java
package org.webfabric.web.servlet;

import org.webfabric.io.Path;

import javax.servlet.http.HttpServletRequest;

public class OriginalServletPath implements Path {
    private final String value;

    public OriginalServletPath(HttpServletRequest request) {
        this.value = (String) request.getAttribute("javax.servlet.include.servlet_path");
    }

    public OriginalServletPath(String value) {
        this.value = value;
    }

    public String value() {
        return value;
    }
}

File: src/org/webfabric/web/servlet/OriginalServletPath.scala
package org.webfabric.web.servlet;

import io.HierarchicalPath
import javax.servlet.http.HttpServletRequest;

class OriginalServletPath(v: String) extends HierarchicalPath(v)

object OriginalServletPath {
  def apply(request: HttpServletRequest):OriginalServletPath = {
    return new OriginalServletPath(request.getAttribute("javax.servlet.include.servlet_path").asInstanceOf[String]);
  }
}
